extensions/azure-cosmos/runtime/src/main/java/io/quarkiverse/azure/cosmos/runtime/CosmosClientProducer.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkiverse.azure.cosmos.runtime;
+
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+
+import io.quarkiverse.azure.core.util.AzureQuarkusIdentifier;
+
+public class CosmosClientProducer {
+
+    @Inject
+    CosmosConfig cosmosConfiguration;
+
+    @Produces
+    public CosmosClient createCosmosClient() {
+        CosmosClientBuilder builder = getBuilder();
+        return null == builder ? null : builder.buildClient();
```

Comment:
In what scenarios might the builder be null?

Reply:
If the extension is disabled, see https://github.com/quarkiverse/quarkus-azure-services/pull/270/files#diff-446df996146b8f10feebd59ae05aed7c682951cfd0cec1cd2c866dc2db2a2e48R31-R33.

Reply:
Great catch.